### PR TITLE
Feature/erc20

### PR DESCRIPTION
**Pull request by davidp94**
Added ERC20 support instead of erc.

Contract is available in `Miximus_ERC20.sol`

### CODE REVIEW

File: snarkWrapper/contracts/Miximus_ERC20.sol [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+        zksnark_verify = Verifier(_zksnark_verify);
+    }
+
+    function deposit (address _erc20Address, bytes32 leaf) payable  {
+        if(!miximusTrees[_erc20Address].instantiated) {
+            miximusTrees[_erc20Address].ercInstance = ERC20(_erc20Address);
+            miximusTrees[_erc20Address].instantiated = true;
+        }
+        require(miximusTrees[_erc20Address].ercInstance.transferFrom(msg.sender, address(this), 1));
+        insert(leaf);
```
Comment (davidp94) [inline snarkWrapper/contracts/Miximus_ERC20.sol]: @barryWhiteHat I don't know how I should modify that in order to have one MerkelTree per `miximusTree`

Comment (davidp94) [reply snarkWrapper/contracts/Miximus_ERC20.sol]: maybe putting the `MTree` structure into the `miximusTree` such as

```
struct MiximusTree {
        ERC20 ercInstance;
        bool instantiated;
        mapping (bytes32 => bool) roots;
        mapping (bytes32 => bool) nullifiers;
        Mtree mtree;
    }
```
?

Comment (barryWhiteHat) [reply snarkWrapper/contracts/Miximus_ERC20.sol]: Yes i think that will work.

**Comment by HarryR** [comment]
Is it possible to split the proving & merkle code into a separate common library which is used by both the Ether and Token contracts?

**Comment by barryWhiteHat** [comment]
@HarryR I am not sure what you mean? Do you mean the reuse the zksnark witness generation for both? Or do you mean remove the `getMerkelProof` function from solidity and implement it in python ?

**Comment by davidp94** [comment]
Moved to #29

**Pull request closed without merging**